### PR TITLE
Massive announcer/BSA overhaul

### PR DESCRIPTION
**Pull request by Gandalf2k15**
## About The Pull Request
Welcome, welcome to city 17.

Alright so this essentially replaces all current default announcer sounds with a few that I found from Tau Ceti. Here's the main chunk of what has changed:

image

All of these announcements have been replaced with Tau Ceti ones, which are much nicer and really do enhance the immersion.  However, some announcements didn't exist, so I added them, they are as follows:

`#define ANNOUNCER_SHUTTLELEFT "announcer_shuttleleft"
#define ANNOUNCER_CARP "announcer_carp"
#define ANNOUNCER_SHUTTLELEFT "announcer_shuttleleft"
#define ANNOUNCER_SHUTTLELEFT "announcer_shuttleleft"
#define ANNOUNCER_VORTEXANOMALIES "announcer_vortexanomalies"
#define ANNOUNCER_CAPTAIN "announcer_captain"
#define ANNOUNCER_MASSIVEBSPACEANOMALIES "announcer_massivebspaceanomalies"
#define ANNOUNCER_PYROANOMALIES "announcer_pyroanomalies"
#define ANNOUNCER_FLUXANOMALIES "announcer_fluxanomalies"
#define ANNOUNCER_GRAVGENON "announcer_gravgenon"
#define ANNOUNCER_GRAVGENOFF "announcer_gravgenoff"
#define ANNOUNCER_GREYTIDE "announcer_greytide"
#define ANNOUNCER_COMMSBLACKOUT "announcer_commsblackout"
#define ANNOUNCER_ELECTRICALSTORM "announcer_electricalstorm"
#define ANNOUNCER_BRANDINTELLIGENCE "announcer_brandintelligence"
#define ANNOUNCER_RADIATIONPASSED "announcer_radiationpasssed"
#define ANNOUNCER_BLUESPACEARTY "announcer_bluespacearty"`

The way you hear announcements has also changed slightly; depending on where you are on the map, you will either hear the sounds at full volume, or low volume. For example: If you're in maintenance, you'll only hear it faintly, where as in the hall, you'll hear it at full volume.

Announcement sounds now have a volume setting in prefs, check it out. 1-100. Default is 60.

## Bluespace Artillery Overhaul

Whew, the BSA has had an overhaul in functionality and is now available all of the time to buy from cargo.

The BSA now requires a powercore(see below) to power it as well as the APC of the room it's in. It has a prefire sequence timer too, takes 30 seconds to reload and 19 seconds to prefire.

It processes power as it prefires, you need to ensure the powernet can support it's prefire sequence for 19 seconds.

https://streamable.com/p5dyry

## Changelog
:cl:
add: A new announcement system has been added to replace the default announcer, many many more events now have custom announcement sounds. The way you will hear announcements will also be different.
add: You can now change announcement volume in preferences.
fix: A lot of announcements not respecting prefs and causing your ears to bleed.
add: The bluespace artillery can now be purchased at any time however is more expensive to run.
/:cl:

### CODE REVIEW

**Comment by GoldenAlpharex** [comment]
It sounds alright, I guess it'll take some getting used to. There's just a few sounds so far that just don't match the same kind of sound level, including the Emergency Access sound notification. I'll come back and edit this comment into a list or whatever as I find them out.

**Comment by GoldenAlpharex** [comment]
Award for the most informative commits :^)

**Comment by FlamingLily** [comment]
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA MY FUCKING EARS

deaf

**Comment by Vizoplez** [comment]
based as fuck

**Comment by YakumoChen** [comment]
I prefer the original female Vox sounds, to be honest

**Comment by GoldenAlpharex** [comment]
> I prefer the original female Vox sounds, to be honest

I do too, in most cases.

**Comment by Gandalf2k15** [comment]
> I prefer the original female Vox sounds, to be honest

They're old, nasty and you can barely make them out. It's time to move on. The past is behind us.

**Comment by YakumoChen** [comment]
That's just your opinion. Vox is ported from Half Life anyway and has lasted us this long without problems.

"The past is behind us" is a garbage thing to say when the station literally still uses floppy disks and PDAs that aren't any higher tech than old Nokia phones.

We still have several dozen sprites in the code and a bunch of atmospherics code that hasn't changed since Exadv created the game. Newer =/= better all the time, sometimes things just work fine and don't need to change.

**Comment by YakumoChen** [comment]
Also atomize your PR, you've got two completely different features bundled into one PR

**Comment by Gandalf2k15** [comment]
> Also atomize your PR, you've got two completely different features bundled into one PR

You're not a maintainer. Stop acting like one. Atomisation is to make it easier for other maintainers to review. I'm the only code maintainer rn. I do my own reviews. Thanks.

**Comment by Gandalf2k15** [comment]
> That's just your opinion. Vox is ported from Half Life anyway and has lasted us this long without problems.
> 
> "The past is behind us" is a garbage thing to say when the station literally still uses floppy disks and PDAs that aren't any higher tech than old Nokia phones.
> 
> We still have several dozen sprites in the code and a bunch of atmospherics code that hasn't changed since Exadv created the game. Newer =/= better all the time, sometimes things just work fine and don't need to change.

Also I never touched the ai vox system?

**Comment by NullDagaf** [comment]
> > Also atomize your PR, you've got two completely different features bundled into one PR
> 
> You're not a maintainer. Stop acting like one. Atomisation is to make it easier for other maintainers to review. I'm the only code maintainer rn. I do my own reviews. Thanks.

He's right tho

**Comment by NullDagaf** [comment]
_also please don't spaghettify set_security_level.dm again @Gandalf2k15, thanks_

**Comment by Gandalf2k15** [comment]
> _also please don't spaghettify set_security_level.dm again @Gandalf2k15, thanks_

I'll rewrite it to use priority announcements since that's what I've refactored, minor announce can suck my cucumber

**Comment by NullDagaf** [comment]
> > _also please don't spaghettify set_security_level.dm again @Gandalf2k15, thanks_
> 
> I'll rewrite it to use priority announcements since that's what I've refactored, minor announce can suck my cucumber

Alright, good.

File: modular_skyrat/modules/bsa_overhaul/code/bsa_cannon.dm [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+
+/obj/machinery/bsa/full/proc/pre_fire(mob/user, turf/bullseye)
+	if(system_state == SYSTEM_READY)
+		priority_announce("BLUESPACE TARGETING PARAMETERS SET, PREIGNITION STARTING... FIRING IN T-20 SECONDS!", "BLUESPACE ARTILLERY", ANNOUNCER_BLUESPACEARTY)
+		alert_sound_to_playing('modular_skyrat/modules/bsa_overhaul/sound/superlaser_prefire.ogg')
+		system_state = SYSTEM_PREFIRE
+		message_admins("[user] has started the fire cycle of [src]!")
+		set_light(5, 5, COLOR_BLUE_LIGHT)
+		START_PROCESSING(SSobj, src)
+		addtimer(CALLBACK(src, .proc/fire, user, bullseye), 15 SECONDS)
```
Comment (Gandalf2k15) [inline modular_skyrat/modules/bsa_overhaul/code/bsa_cannon.dm]:
```suggestion
		addtimer(CALLBACK(src, .proc/fire, user, bullseye), 15 SECONDS)
		START_PROCESSING(SSobj, src)
```

**Comment by Gandalf2k15** [comment]
Full merging now to prevent future merge conflicts. Testing phase complete.

**Comment by IgiariValkyr** [comment]
Put the video back up. We can't fucking see how to build it.

**Pull request merged**